test/rules/no-invalid-this/enabled/test.ts.lint

```diff
@@ -1,3 +1,21 @@
+export function f<T>(this: Observable<T>): Observable<T> {
+    const nestedFunction = (this) => {
```

Comment:
arrow functions cannot have a `this` parameter
We don't need to worry about it, because TypeScript shows an error. But we shouldn't use it in any test